frontends/web/src/routes/account/add/add.tsx

```diff
@@ -206,96 +179,80 @@ class AddAccount extends Component<Props, State> {
-    return (
-      <div className="contentWithGuide">
-        <div className="container">
-          <div className="innerContainer scrollableContainer">
-            <Header title={<h2>{t('manageAccounts.title')}</h2>} />
-            <div className="content larger isVerticallyCentered">
-              <form
-                className={`${styles.manageContainer} box larger flex flex-column flex-between`}
-                onSubmit={this.next}>
-                <div className="text-center">
-                  {t('addAccount.title')}
-                  <h1 className={styles.title}>{titleText}</h1>
-                </div>
-                <div className="row" hidden={!errorMessage}>
-                  <Message type="warning">{errorMessage}</Message>
-                </div>
-                <div className="row">
-                  {this.renderContent()}
-                </div>
-                <div className="row">
-                  <Steps current={currentStep}>
-                    <Step key="select-coin" hidden={this.onlyOneSupportedCoin()}>
-                      {t('addAccount.selectCoin.step')}
-                    </Step>
-                    <Step key="choose-name">
-                      {t('addAccount.chooseName.step')}
-                    </Step>
-                    <Step key="success">
-                      {t('addAccount.success.step')}
-                    </Step>
-                  </Steps>
-                </div>
-                <div className="row flex flex-row flex-between m-bottom" style={{ flexDirection: 'row-reverse' }}>
-                  <Button
-                    disabled={
-                      (step === 'select-coin' && coinCode === 'choose')
-                                        || (step === 'choose-name' && (accountName === '' || adding))
-                    }
-                    primary
-                    type="submit">
-                    {nextButtonText}
-                  </Button>
-                  <Button
-                    onClick={this.back}
-                    hidden={step === 'success'}
-                    transparent>
-                    {t('button.back')}
-                  </Button>
-                  <Button
-                    onClick={() => this.setState({
-                      accountCode: undefined,
-                      accountName: '',
-                      coinCode: 'choose',
-                      errorMessage: undefined,
-                      step: 'select-coin',
-                    }, this.startProcess)}
-                    hidden={step !== 'success'}
-                    transparent>
-                    Add another account
-                  </Button>
-                </div>
-              </form>
-            </div>
+  const handleAddAnotherAccount = () => {
+    setAccocuntCode(undefined);
+    setAccountName('');
+    setCoinCode('choose');
+    setErrorMessage(undefined);
+    setStep('select-coin');
+  };
```

Comment:
I'm not super clear w/ the suggestion. Is it something like this?

```js
  useEffect(() => {
    if (step === 'choose-name') {
      inputRef.current?.focus();
      startProcess();
    }
  }, [step]);
```

If so, this won't work because in the `choose-name` step it'll re-start the process because of `startProcess()` and user will be stuck in a loop. But I'm not sure if that's what you meant.

Or is it something like this:


```js
  useEffect(() => {
    startProcess();
  }, []);

  useEffect(() => {
    if (step === 'select-coin') {
      startProcess();
    }
  }, [step]);
```
I think we can even simplify it to:

```js
  useEffect(() => {
    if (step === 'select-coin') {
      startProcess();
    }
  }, [step]);
```

This seems to work okay. tThe reason I did it the other way was because I was doing the equivalent of the class component as I don't want to miss anything since in the class component it's the callback of **all** of these states changes, and not only the `step` state:

``js
this.setState({
                      accountCode: undefined,
                      accountName: '',
                      coinCode: 'choose',
                      errorMessage: undefined,
                      step: 'select-coin',
                    }, this.startProcess)}
```
but.. it seems like watching the `step` only when it's `select-coin` is sufficient..?

Reply:
Actually, I just noticed there's a slight difference in behaviour when we only watch for `step === select-coin`.  This is the minimum that we can have to maintain the same behaviour as the one in master: 

```js
  useEffect(() => {
    if (step === 'select-coin' && coinCode === 'choose') {
      startProcess();
    }
  }, [step, coinCode]);
```

I'll make this change for now.. but a friendly reminder that isn't equivalent of what we have in the class component and we may miss something (like I just did 😄, although small chance).